Misc fixes for Solaris and Darwin
Removal of hardcoded search paths

--- configure
+++ configure
@@ -526,7 +526,7 @@
       mksharedlib="$flexlink"
       mkmaindll="$flexlink -maindll"
       shared_libraries_supported=true;;
-    *-*-linux-gnu|*-*-linux|*-*-freebsd[3-9]*|*-*-openbsd*|*-*-netbsd*|*-*-gnu*)
+    *-*-solaris2*|*-*-linux-gnu|*-*-linux|*-*-freebsd[3-9]*|*-*-openbsd*|*-*-netbsd*|*-*-gnu*)
       cmxs="cmxs"
       sharedcccompopts="-fPIC"
       mksharedlib="$bytecc -shared"
@@ -653,7 +653,7 @@
   powerpc*-*-linux*)            arch=power; model=ppc; system=elf;;
   powerpc-*-netbsd*)            arch=power; model=ppc; system=elf;;
   powerpc-*-rhapsody*)          arch=power; model=ppc; system=rhapsody;;
-  powerpc-*-darwin*)            arch=power; system=rhapsody
+  powerpc-*-darwin*)            arch=power; system=macosx
                                 if $arch64; then model=ppc64; else model=ppc; fi;;
   arm*-*-linux*)                arch=arm; system=linux;;
   arm*-*-gnu*)                  arch=arm; system=gnu;;
@@ -665,7 +665,8 @@
   x86_64-*-freebsd*)            arch=amd64; system=freebsd;;
   x86_64-*-netbsd*)             arch=amd64; system=netbsd;;
   x86_64-*-openbsd*)            arch=amd64; system=openbsd;;
-  x86_64-*-darwin9.5)           arch=amd64; system=macosx;;
+  x86_64-*-darwin*)             arch=amd64; system=macosx;;
+  x86_64-*-solaris*)            arch=amd64; system=solaris;;
 esac
 
 # Some platforms exist both in 32-bit and 64-bit variants, not distinguished
@@ -728,7 +729,10 @@
   hppa,*,*)         as='as';
                     aspp='gcc -traditional -c';;
   i386,*,solaris)   as='as'
-                    aspp='/usr/ccs/bin/as -P';;
+                    case "$cc" in
+                      gcc*) aspp='gcc -c';;
+                         *) aspp='as -P';;
+                    esac;;
   i386,*,*)         as='as'
                     aspp='gcc -c';;
   ia64,*,*)         as='as -xexplicit'
@@ -1150,8 +1154,8 @@
 
 if test "$pthread_wanted" = "yes"; then
   case "$host" in
-    *-*-solaris*)  pthread_link="-lpthread -lposix4"
-                   pthread_caml_link="-cclib -lpthread -cclib -lposix4";;
+    *-*-solaris*)  pthread_link="-pthread"
+                   pthread_caml_link="-cclib -pthread";;
     *-*-freebsd*)  pthread_link="-pthread"
                    pthread_caml_link="-cclib -pthread";;
     *-*-openbsd*)  pthread_link="-pthread"
@@ -1351,27 +1355,11 @@
 dbm_link="not found"
 use_gdbm_ndbm=no
 
-for dir in /usr/include /usr/include/db1 /usr/include/gdbm; do
+for dir in $prefix/include/gdbm; do
   if test -f $dir/ndbm.h; then
     dbm_include=$dir
-    if sh ./hasgot dbm_open; then
-      dbm_link=""
-    elif sh ./hasgot -lndbm dbm_open; then
-      dbm_link="-lndbm"
-    elif sh ./hasgot -ldb1 dbm_open; then
-      dbm_link="-ldb1"
-    elif sh ./hasgot -lgdbm dbm_open; then
+    if sh ./hasgot -lgdbm dbm_open; then
       dbm_link="-lgdbm"
-    elif sh ./hasgot -lgdbm_compat -lgdbm dbm_open; then
-      dbm_link="-lgdbm_compat -lgdbm"
-    fi
-    break
-  fi
-  if test -f $dir/gdbm-ndbm.h; then
-    dbm_include=$dir
-    use_gdbm_ndbm=yes
-    if sh ./hasgot -lgdbm_compat -lgdbm dbm_open; then
-      dbm_link="-lgdbm_compat -lgdbm"
     fi
     break
   fi
@@ -1414,18 +1402,11 @@
   tcl_version=''
   tcl_version=`sh ./runtest $tk_defs $tk_x11_include tclversion.c`
   for tk_incs in \
-    "-I/usr/local/include" \
-    "-I/usr/include" \
-    "-I/usr/local/include/tcl8.5 -I/usr/local/include/tk8.5" \
-    "-I/usr/include/tcl8.5 -I/usr/include/tk8.5" \
-    "-I/usr/local/include/tcl8.4 -I/usr/local/include/tk8.4" \
-    "-I/usr/include/tcl8.4 -I/usr/include/tk8.4" \
-    "-I/usr/local/include/tcl8.3 -I/usr/local/include/tk8.3" \
-    "-I/usr/include/tcl8.3 -I/usr/include/tk8.3" \
-    "-I/usr/local/include/tcl8.2 -I/usr/local/include/tk8.2" \
-    "-I/usr/include/tcl8.2 -I/usr/include/tk8.2" \
-    "-I/sw/include" \
-    "-I/usr/pkg/include"
+    "-I$prefix/include" \
+    "-I$prefix/include/tcl8.5 -I$prefix/include/tk8.5" \
+    "-I$prefix/include/tcl8.4 -I$prefix/include/tk8.4" \
+    "-I$prefix/include/tcl8.3 -I$prefix/include/tk8.3" \
+    "-I$prefix/include/tcl8.2 -I$prefix/include/tk8.2"
   do if test -z "$tcl_version"; then
     tk_defs="$tk_incs"
     tcl_version=`sh ./runtest $tk_defs $tk_x11_include tclversion.c`
@@ -1469,17 +1450,6 @@
   elif sh ./hasgot $tk_libs -ltcl$tclmaj$tclmin $tkauxlibs Tcl_DoOneEvent
   then
     tk_libs="$tk_libs -ltk$tkmaj$tkmin -ltcl$tclmaj$tclmin $dllib"
-  elif sh ./hasgot -L/sw/lib $tk_libs -ltcl$tclmaj.$tclmin $tkauxlibs \
-                   Tcl_DoOneEvent
-  then tk_libs="-L/sw/lib -ltk$tkmaj.$tkmin -ltcl$tclmaj.$tclmin $dllib"
-  elif sh ./hasgot -L/usr/pkg/lib $tk_libs $tk_x11_libs \
-        -ltk$tkmaj$tkmin -ltcl$tclmaj$tclmin -lpthread $tkauxlibs \
-                   Tcl_DoOneEvent
-  then
-    case "$host" in
-      *-*-*bsd*) tk_libs="-R/usr/pkg/lib -L/usr/pkg/lib $tk_libs $tk_x11_libs -ltk$tkmaj$tkmin -ltcl$tclmaj$tclmin -lpthread $tkauxlibs";;
-      *) tk_libs="-L/usr/pkg/lib $tk_libs $tk_x11_libs -ltk$tkmaj$tkmin -ltcl$tclmaj$tclmin -lpthread $tkauxlibs";;
-    esac
   else
     echo "Tcl library not found."
     has_tk=false
@@ -1489,19 +1459,6 @@
 if test $has_tk = true; then
   if sh ./hasgot $tk_libs $tk_x11_libs $tkauxlibs Tk_SetGrid; then
     echo "Tcl/Tk libraries found."
-  elif sh ./hasgot -L/sw/lib $tk_libs $tk_x11_libs $tkauxlibs Tk_SetGrid; then
-    case "$host" in
-      *-*-*bsd*) tk_libs="-R/sw/lib -L/sw/lib $tk_libs";;
-      *) tk_libs="-L/sw/lib $tk_libs";;
-    esac
-    echo "Tcl/Tk libraries found."
-  elif sh ./hasgot -L/usr/pkg/lib $tk_libs $tk_x11_libs $tkauxlibs \
-                   Tk_SetGrid; then
-    case "$host" in
-      *-*-*bsd*) tk_libs="-R/usr/pkg/lib -L/usr/pkg/lib $tk_libs";;
-      *) tk_libs="-L/usr/pkg/lib $tk_libs";;
-    esac
-    echo "Tcl/Tk libraries found."
   else
     echo "Tcl library found."
     echo "Tk library not found."
